spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/AbstractInstancePersistingZKDeployer.java

```diff
@@ -131,4 +137,60 @@ public void deleteAll() {
 	 */
 	protected abstract I makeInstance(D definition);
 
+	/**
+	 * Deploy the deployment unit with the given name.
+	 *
+	 * @param deploymentUnitName the deployment unit name
+	 */
+	protected final void deployResource(String deploymentUnitName) {
+		try {
+			deploymentHandler.deploy(deploymentUnitName);
+		}
+		catch (Exception e) {
+			throw new DeploymentException(deploymentUnitName);
```

Comment:
Need to pass the original Exception, especially given the same class "DeploymentException" is used in the undeploy case. An additional message adding context would be welcome too